### DISCUSSION TITLE
Trendline with equation

### DISCUSSION THREAD

**Discussion by brianjoubert** [adamerose/pandasgui#185] [Q&A] [answered]
Hi,

I would like to add a trendline (with the equation on the plot if possible) on a selected set of datapoints. I tried to add it as custom Kwarg like this

Kwarg Name        Kwarg Value
trendline                   "ols"

but it doesn't appear anywhere.

Am I missing something?

Thanks a lot

**Comment by adamerose** [comment, marked as answer]
Hi @brianjoubert , I just tried it on a scatter plot and it works for me. You may have been using a graph that doesn't support the trendline argument? Please check this reference https://plotly.com/python-api-reference/generated/plotly.express.scatter.html#plotly.express.scatter, and if that isn't the reason, create a minimum reproducible example that I can look into.
![image](https://user-images.githubusercontent.com/10884874/149140939-feddfc60-ad9b-4b51-be63-1e56c918f06d.png)


Or are you saying the trendline works but you're just missing the equation showing on the plot? That would be a new feature request to make on the Plotly Express repo. In the meantime you would need to calculate it yourself and manually create a textbox on the Plotly figure

**Reply by brianjoubert** [reply]
Thanks a lot for your reply. When I saw your screenshot I immediately understood I was not running the same version. I stupidly installed it from conda without checking the version. Now installed it with pip and it does work :)

Thanks a lot

**Reply by adamerose** [reply]
> Thanks a lot for your reply. When I saw your screenshot I immediately understood I was not running the same version. I stupidly installed it from conda without checking the version. Now installed it with pip and it does work :)
> 
> Thanks a lot

Ah yeah that makes sense, unfortunately I've been not maintaining the conda repo very well since I don't use it myself, I think I may remove it from the readme until I have the bandwidth

**Reply by brianjoubert** [reply]
The only thing I still don't manage to do is to access the data from the trendline as I would do with something like this
Trendline_results = px.get_trendline_results(fig)

Is it possible to do such things or the only solution is to export the whole code and do it in jupyterlab (or equivalent). Generally speaking is there a way to use other libraries such as Numpy to do some operations on the database or it is just out of the scope of PandasGUI?

**Reply by adamerose** [reply]
> The only thing I still don't manage to do is to access the data from the trendline as I would do with something like this Trendline_results = px.get_trendline_results(fig)
> 
> Is it possible to do such things or the only solution is to export the whole code and do it in jupyterlab (or equivalent). Generally speaking is there a way to use other libraries such as Numpy to do some operations on the database or it is just out of the scope of PandasGUI?

It's not covered in the docs and subject to change in later versions, but you can access the fig object like this

`fig = gui.store.data['your_df_name'].dataframe_explorer.grapher.fig`
